text/0052-resource-importing-support.md

```diff
@@ -0,0 +1,272 @@
+# Support for importing existing resources into a CDK stack
+
+* **Original Author(s):**: @tomas-mazak
+* **Tracking Issue**: #52
+* **API Bar Raiser**: @rix0rrr
+
+Allow users to import resources, that were created manually or by different
+means, into CDK stacks.
+
+## Working Backwards - README
+
+```
+cdk import [STACK]
+
+Import existing resource(s) into the given STACK
+
+Options:
+[... standard CDK CLI and CDK CLI `deploy` options ...]
+  -o, --create-resource-mapping  If specified, CDK will generate a mapping of
+                                 existing physical resources to CDK constructs to
+                                 be imported as. The mapping will be written in
+                                 the given file path. No actual import operation
+                                 will be performed                      [string]
+  -m, --resource-mapping         If specified, CDK will use the given file to
```

Comment:
Do you have a particular use case  in mind for these parameters? I see how it *might* be useful, but if this is with an eye towards a non-permissioned CI/CD workflow, just adding these flags is not going to be enough by itself.

Can you describe in the README a use case a user might find themselves in, in which they want this feature, and what the workflow looks like in that case?

Reply:
What's the structure of the resource mapping file? 

If this is going to be useful in an unattended scenario, it will have to be a 2-level map I think, right?

```
{ stack -> { logicalId -> physicalId } }
```

?

If so, you probably want to be able to combine the outputs of multiple `import` calls into the same output file.

Reply:
I have updated this as requested.